### DISCUSSION TITLE
Can I assign a virtual chassis member variable to an interface?

### DISCUSSION THREAD

**Discussion by CupcakeMunster** [netbox-community/netbox#10366] [Q&A] [answered]
I know all the interfaces appear on the master device in a virtual chassis. Can I define an interface name like `xe-{member}/0/0` so that when I'm looking at my interfaces on the master device I can see `xe-0/0/0` for itself and `xe-1/0/0` for it's partner device?

**Comment by mtinberg** [comment, marked as answer]
IIUC there is a misunderstanding here, if you have a Device Type with Interface xe-0/0/0 defined on it, then create an instance of this as switch-1 and another instance as switch-2, both will have a xe-0/0/0 interface defined, if you combine them with a Virtual Chassis into a stack then you'll see two (2) xe-0/0/0 interfaces when you look at the whole stack and you will want to navigate to the Device record for switch-2 itself and bulk rename them, using the Web UI or a Script or REST client app.

I believe there was a feature request to allow the interfaces to be templated like "xe-{position}/0/0", as was done for Module Bays, so the number gets replaced with the Virtual Chassis position when the db records get created, but I wasn't sure if that feature was done yet or still in progress.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: CupcakeMunster ***@***.***>
Sent: Wednesday, September 14, 2022 12:56 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Can I assign a virtual chassis member variable to an interface? (Discussion #10366)


I know all the interfaces appear on the master device in a virtual chassis. Can I define an interface name like xe-{member}/0/0 so that when I'm looking at my interfaces on the master device I can see xe-0/0/0 for itself and xe-1/0/0 for it's partner device?

**Reply by CupcakeMunster** [reply]
Bulk renaming actually works nicely. Thanks for pointing that out. I just wanted to make sure I wasn't missing some functionality that would make my life easier down the road but bulk renaming only took me a few seconds.
